lib/elixir/src/elixir_def.erl

```diff
@@ -246,7 +246,7 @@ store_definition(Check, Kind, Meta, Name, Arity, File, Module, Defaults, Clauses
   {MaxDefaults, FirstMeta} =
     case ets:lookup(Set, {def, Tuple}) of
       [{_, StoredKind, StoredMeta, StoredFile, StoredCheck, {StoredDefaults, LastHasBody, LastDefaults}}] ->
-        check_valid_kind(Meta, File, Name, Arity, Kind, StoredKind),
+        check_valid_kind(Meta, File, Name, Arity, Kind, StoredKind, StoredMeta),
```

Comment:
You need to pass StoredFile too and use that in the error message too.

Reply:
Sorry. Fixed